app/README.md

```diff
@@ -8,12 +8,92 @@ This is a JavaScript application which uses (in `dependencies` of its `package.j
 `@hawtio/react` package.
 
 While `@hawtio/react` is an NPM package bundled with `tsup` (`esbuild` + `rollup`), this application is bundled
-using `webpack` to produce code running inside browser.
+using `webpack` to produce code running inside the browser.
 
-This application not only shortens the dev cycle (change `@hawtio/react` > see results in the browser), it also
-shows how other applications ([Hawtio Standalone](https://github.com/hawtio/hawtio/tree/4.x/console) being one of them)
+This application not only shortens the dev cycle (change `@hawtio/react` - see results in the browser), it also
+shows how other applications ([Hawtio Standalone](https://github.com/hawtio/hawtio/tree/4.x/console) being one of them):
 
 - should be structured
 - should use `webpack` and configure [Module Federation](https://webpack.js.org/concepts/module-federation/)
 - should initialize and bootstrap `hawtio` classes
-- should render initial React application
+- should render main React component (`<Hawtio>`)
+
+## dependencies, dev dependencies, peer dependencies
+
+This `app` NPM package is bundled using Webpack and produces an _application_ that runs in browser. This is _not_ a published NPM packages and will never be a dependency of any other NPM project.
+
+In terms of dependencies we have to remember few things:
+
+- This `app` definitely depends on `react`, `react-dom` and Patternfly libraries and `@hawtio/react` package
+- This `app` transitively (implicitly) depends on other libraries (dependencies of the above)
+- `webpack.config.cjs` configuration of this `app` uses some _shared modules_ (with the help of [Module Federation mechanism](https://webpack.js.org/concepts/module-federation/)).
+
+Application bundled with Webpack should not have any `peerDependencies`, because it's not used by other packages in their `dependencies`.
+
+This `app` should declare a `dependency` only if it's used directly in its code (for example `src/bootstrap.tsx` imports React functions, so `react` should be a direct dependency). This `app` should not add Jolokia to it's `dependencies` - Webpack will bundle it together with `@hawtio/react` without problems.
+
+If there's a need to select specific version of a transitive package (for example due to security reasons or known CVE), such dependency should be added to `resolutions` config in `package.json`.
```

Comment:
- specific version ---> a specific version
- such dependency ---> such a dependency
- `resolutions` config ---> the `resolutions` property

Reply:
true, but if there's a CVE in transitive dependency I think we can use `resolutions` instead of saying "this is our 1st level dependency".
Scanners just scan node_modules/ which may have nested node_modules/ and so on...